extensions/Datatype.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+trait Datatype {
+
+    /**
+     * Datatypes
+     * Datatypes of CSV data-columns
+     *
+     * @access public
+     * @var array
+     */
+    public $data_types = [];
+
+    /**
+     * Check data type
+     * Check for possible data types for one field value string.
+     *
+     * @access private
+     *
+     * @param string $value cell value
+     *
+     * @return string
+     */
+    private function getDatatypeFromString($value) {
+        $value = trim((string) $value);
+
+        if (empty($value)) {
+            return 'unknown';
+        }
+
+        if (preg_match('/^(?i:true|false)$/', $value)) {
+            return 'boolean';
+        }
+
+        if (preg_match('/^[-+]?[0-9]\d*$/', $value)) {
+            return 'integer';
+        }
+
+        if (preg_match('/^[+-]?([0-9]*[.])?([0-9]|[.][0-9])+$/', $value)) {
+            return 'float';
+        }
+
+        if ((bool) strtotime($value)) {
+            return 'date';
+        }
+
+        return 'string';
+    }
+
+    /**
+     * Check data type for one column.
+     * Check for most commonly data type for one column.
+     *
+     * @access private
+     *
+     * @param  [array] $datatypes
+     *
+     * @return [string|false]
+     */
+    private function getMostFrequentDataypeForColumn($datatypes) {
+        if (isset($datatypes['unknown'])) {
+            unset($datatypes['unknown']);
+        }
+
+        $typesFreq = array_count_values($datatypes);
+        arsort($typesFreq);
+        reset($typesFreq);
+
+        return key($typesFreq);
+
+    }
+
+    /**
+     * Check data type foreach Column
+     * Check data type for each column and returns the most commonly.
+     *
+     * @access public
+     *
+     * @return [array|bool]
+     */
+    public function getDatatypes() {
+        if (empty($this->data)) {
+            $this->data = $this->parse_string();
+        }
+
+        $result = [];
+        foreach ($this->titles as $cName) {
+            $column = array_column($this->data, $cName);
+            $cDatatypes = array_map([$this, 'getDatatypeFromString'], $column);
+
+            $result[$cName] = $this->getMostFrequentDataypeForColumn($cDatatypes);
+        }
+
+        $this->data_types = $result;
+
+        return !empty($this->data_types) ? $this->data_types : FALSE;
```

Comment:
Code style: the rest of the code base uses lower-case constants. You can use [this PhpStorm code style archive](https://github.com/parsecsv/parsecsv-for-php/issues/101#issuecomment-355733791) to speed things up. Thanks!

Reply:
done.... and I miss the GitLab `Resolve`-Button here in GitHub for this situation 😟